### PR TITLE
pep 484: explicit reexport as intended

### PR DESCRIPTION
**Pull request by hauntsaninja**
See discussion on typing-sig.
This doesn't take care of some third_party libraries, will follow up on
those.

### CODE REVIEW

File: stdlib/3/os/__init__.pyi [hunk 1]
```diff
@@ -268,7 +270,9 @@ class stat_result:
     st_type: int
 
 if sys.version_info >= (3, 6):
-    from builtins import _PathLike as PathLike  # See comment in builtins
+    from builtins import _PathLike
+
```
Comment (gvanrossum) [inline stdlib/3/os/__init__.pyi:274]: I don't think this blank line helps readability.

File: stdlib/3/platform.pyi [hunk 2]
```diff
@@ -3,7 +3,9 @@
 import sys
 
 if sys.version_info < (3, 9):
-    from os import devnull as DEV_NULL
+    import os
+
```
Comment (gvanrossum) [inline stdlib/3/platform.pyi:7]: No blank line here.

File: third_party/2/six/__init__.pyi [hunk 3]
```diff
@@ -5,7 +5,9 @@ import typing
 import unittest
 from __builtin__ import unichr as unichr
 from functools import wraps as wraps
-from StringIO import StringIO as BytesIO, StringIO as StringIO
+from StringIO import StringIO as StringIO
+
```
Comment (gvanrossum) [inline third_party/2/six/__init__.pyi:9]: No blank line.

**Review by gvanrossum — COMMENTED** [review]
Thanks! I found a few blank lines that strike me as excessive (we're not following PEP 8 on purpose).

**Comment by hauntsaninja** [comment]
Typeshed has surrendered itself to black, which mandates those blank lines

**Comment by gvanrossum** [comment]
Oooh, that last failing test is a fun one. The code flow through mypy is entirely different when defining a type alias instead of renaming on import.

**Review by srittau — APPROVED** [review]
I'll tentatively approve this, pending a solution for the mypy failures. I think it's a good idea that the re-export ambiguity is being discussed and fixed.

**Comment by JelleZijlstra** [comment]
I haven't checked but I think you need to alias generics with something like `PathLike = _PathLike[T]`, possibly with a more specific typevar.

**Comment by hauntsaninja** [comment]
I think this just works and needs builtins._PathLike to not be hardcoded in mypy. Going attempt the merge dance.

**Comment by jakebailey** [comment]
Yes; I was just about to say that I think it'd be weird to have to add typevars to emulate behavior that would have previously worked in the "`as` assignment", especially if we're expecting the PEP clarification to go through and existing stubbers/annotators to fix up their code.

**Pull request merged**